litex/tools/litex_json2dts_linux.py

```diff
@@ -44,18 +44,35 @@ def generate_dts(d, initrd_start=None, initrd_size=None, polling=False):
     if initrd_size is None:
         initrd_size = default_initrd_size
 
+    if initrd == "enabled" or initrd is None:
+        initrd_enabled = True
+    elif initrd == "disabled":
+        initrd_enabled = False
+    else:
+        initrd_enabled = True
+        initrd_size = os.path.getsize(initrd)
+
+    if root_device is None:
+        root_device = "ram0"
+
     dts += """
         chosen {{
-            bootargs = "mem={main_ram_size_mb}M@0x{main_ram_base:x} rootwait console=liteuart earlycon=sbi root=/dev/ram0 init=/sbin/init swiotlb=32";
-            linux,initrd-start = <0x{linux_initrd_start:x}>;
-            linux,initrd-end   = <0x{linux_initrd_end:x}>;
-        }};
-""".format(
+            bootargs = "mem={main_ram_size_mb}M@0x{main_ram_base:x} {console} {rootfs} init=/sbin/init swiotlb=32";""".format(
```

Comment:
Unless we have a bug, I don't think we need to force the number of IO TLB slabs (`swiotlb=32`). If it is required, we should open a bug against the CPU that needs it and document it here.

Reply:
Yes, I don't think its needed.  But I am not running the risc-v setup to know what they are doing.